build.psm1

```diff
@@ -2320,6 +2327,55 @@ function Clear-PSRepo
     }
 }
 
+
+# Install PowerShell modules from a git Repo such as PSL-Pester
+function Restore-GitModule
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Uri,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Destination,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$CommitSha
+        )
+
+        log 'Restoring module from git repo:'
+        log ("Name='{0}', Destination='{1}', Uri='{2}', CommitSha='{3}'" -f $Name, $Destination, $Uri, $CommitSha)
+
+        $clonePath = Join-Path -Path $Destination -ChildPath $Name
+        if(Test-Path $clonePath)
+        {
+            remove-Item -Path $clonePath -recurse -force
+        }
+
+        $null = Start-NativeExecution {git clone --quiet $uri $clonePath}
+        Push-location
+        try {
+            Set-Location $clonePath
```

Comment:
Minor: maybe use `push-location $clonePath` to combine `push-location` and `set-location $clonePath`